src/TransferMsg.chpl

```diff
@@ -0,0 +1,883 @@
+module TransferMsg
+{
+    use MultiTypeSymbolTable;
+    use MultiTypeSymEntry;
+    use Message;
+    use Reflection;
+    use CTypes;
+    use ZMQ;
+    use List;
+    use GenSymIO;
+    use Map;
+    use ServerErrors;
+    use ServerConfig;
+    use ServerErrorStrings;
+
+    use SegmentedString;
+
+    proc sendDataFrameSetupInfo(port:string, numColumns: int, elements: string) throws {
+      var context: Context;
+      var socket = context.socket(ZMQ.PUSH);
+      socket.bind("tcp://*:"+port);
+
+      // first, send the number of columns
+      socket.send(numColumns);
+
+      // next, send the obj type string
+      socket.send(elements);
+    }
+
+    proc receiveDataFrameSetupInfo(hostname: string, port:string) throws {
+      var context: Context;
+      var socket = context.socket(ZMQ.PULL);
+      if hostname == here.name then
+        socket.connect("tcp://localhost:"+port:string);
+      else
+        socket.connect("tcp://"+hostname+":"+port:string);
+      var numColumns = socket.recv(int);
+      
+      var objString = socket.recv(string);
+      var objNames = objString.split(" ");
+
+      return (numColumns, objNames);
+    }
+    
+    proc sendDataFrameMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab): MsgTuple throws {
+      param pn = Reflection.getRoutineName();
+      
+      var hostname = msgArgs.getValueOf("hostname");
+      var port = msgArgs.getValueOf("port");
+      
+      var numColumns = msgArgs.get("size").getIntValue();
+      var eleList = msgArgs.get("columns").getList(numColumns);
+
+      var localeCount = receiveLocaleCount(hostname, port:string);
+      
+      sendDataFrameSetupInfo(port:string, numColumns, eleList: string);
+
+      var colNames = "";
+      for ele in eleList {
+        var ele_parts = ele.split("+");
+        sendColumnName(port, ele_parts[1]);
+        if ele_parts[0] == "Categorical" {
```

Comment:
It would be good if we use the `ObjType` enum here. I think we will need to add some things to it for that.